azure-samples/src/main/java/com/microsoft/azure/management/compute/samples/VirtualMachinesFromCustomImageAndSpecializedVHD.java

```diff
@@ -0,0 +1,223 @@
+                        .withNewResourceGroup(rgName)
+                        .withNewPrimaryNetwork("10.0.0.0/28")
+                        .withPrimaryPrivateIpAddressDynamic()
+                        .withNewPrimaryPublicIpAddress(publicIpDnsLabel)
+                        .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS)
+                        .withRootUserName(userName)
+                        .withPassword(password)
+                        .withSize(VirtualMachineSizeTypes.STANDARD_D3_V2)
+                        .create();
+
+                System.out.println("Created a Linux VM: " + linuxVM.id());
+                Utils.print(linuxVM);
+
+                System.out.println("Please SSH into the VM [" + linuxVM.getPrimaryPublicIpAddress().fqdn() + "]");
+                System.out.println("and run 'sudo waagent -deprovision+user' to prepare it for capturing");
+                System.out.println("after that please 'Enter' to continue.");
+                System.in.read();
+
+                //=============================================================
+                // Deallocate the virtual machine
+                System.out.println("Deallocate VM: " + linuxVM.id());
+
+                linuxVM.deallocate();
+
+                System.out.println("Deallocated VM: " + linuxVM.id() + "; state = " + linuxVM.powerState());
+
+                //=============================================================
+                // Generalize the virtual machine
+                System.out.println("Generalize VM: " + linuxVM.id());
+
+               // linuxVM.generalize();
+
+                System.out.println("Generalized VM: " + linuxVM.id());
+
+                //=============================================================
+                // Capture the virtual machine
+                System.out.println("Capturing VM: " + linuxVM.id());
+
+                String capturedResultJson = linuxVM.capture("capturedvhds", "img", true);
+
+                System.out.println("Captured VM: " + linuxVM.id());
+
+                //=============================================================
+                // Create a Linux VM using captured image
+                String capturedImageUri = extractCapturedImageUri(capturedResultJson);
+
+                System.out.println("Creating a Linux VM using captured image - " + capturedImageUri);
+
+                VirtualMachine linuxVM2 = azure.virtualMachines().define(linuxVmName2)
+                        .withRegion(Region.US_EAST)
+                        .withExistingResourceGroup(rgName)
+                        .withNewPrimaryNetwork("10.0.0.0/28")
+                        .withPrimaryPrivateIpAddressDynamic()
+                        .withoutPrimaryPublicIpAddress()
+                        .withStoredLinuxImage(capturedImageUri) // Note: A URI to generalized VHD is also considered as stored image
+                        .withRootUserName(userName)
```

Comment:
Is it mandatory to use root/admin username/password? What if we want to keep the VM as-is without any customization, including admin password?